### ISSUE TITLE
correctly report assignment to imported variable error in TypeScript

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#1141]
The following is valid TypeScript if the imported `x` is a type-only declaration (e.g. an `interface`):
```typescript
import {x} from './mod';
x = 42;
var x;
```

The following is invalid TypeScript even if the imported `x` is a runtime declaration:
```typescript
import {x} from './mod';
x = 42;
```

We should fix quick-lint-js's variable analyzer to handle both of these cases properly.
